### PR TITLE
Combined classic ItemsSketchSortedView and KllItemsSketchSortedView.

### PR DESCRIPTION
**Pull request by leerho**
Combined classic ItemsSketchSortedView and KllItemsSketchSortedView.
Merged into one shared class, eliminating a big chunk of duplicate code.

By doing this it eliminated the need to copy all the Partitioning code from one to the other.

Also, I was able to eliminate two proxy classes by making the ItemsSketchSortedView constructor public.

### CODE REVIEW

**Comment by leerho** [comment]
While working on this I noticed that the toString(...) method in the
Classic Items sketch was essentially broken.
- The most serious issue is that it made a silent assumption that the T object input by the user overrides the Object.toString() with a human readable and meaningful string.  Without that, the method outputs the very long and not meaningful string generated  by Object, such as: `org.apache.datasketches.quantiles.ItemsSketchTest$BadItem@5e955596`.
- There are many cases where the user may not be able to modify the T object in order to add a useful toString() method.  The user's only option would be to wrap each input object in a class with the toString method.  Yikes!  That would kill performance! 
- Also, the method only provided readable output for very small sketches because of the way the output was structured.  
- This PR fixes the output structuring so that both the KLL Items SV and the Classic Items SV appear almost identical.  In a future PR I want to replicate this same structuring to the KLL Floats and Doubles sketches toString() methods behave the same.
- What is not yet fixed is the fundamental problem is that the Classic Items Sketch constructors only require the Class\<T\> parameter, while the KLL Items Sketch constructors require an implementation of ArrayOfItems\<T\>.  This solves the problem because the AOI class includes a toString method, which can be applied only when required.
- It would be possible to require an AOI class as a parameter to toString(...), but this throws a wrench into having a consistent API across all the quantile sketches.
- The repercussions of this are not simple. I will post a discussion on this issue in Slack.

File: src/main/java/org/apache/datasketches/quantiles/ItemsUtil.java [hunk 1]
```diff
@@ -79,76 +80,117 @@
...
+        final String row = String.format("%7s %7s %7s   %s", i, lvlStr, validLvl, items[i]);
+        sb.append(row).append(LS);
+      }
     }
+    sb.append("### END DATA DETAIL").append(LS);
     return sb.toString();
   }
 
+  private static boolean getValidFromIndex(final int levelNum, final long bitPattern, final int index,
+      final int k, final int bbCount) {
```
[the remark this answers is not in the document]
Comment (leerho) [reply src/main/java/org/apache/datasketches/quantiles/ItemsUtil.java:183]: will fix.

File: src/test/java/org/apache/datasketches/kll/KllItemsSketchTest.java [hunk 2]
```diff
@@ -740,6 +730,35 @@
...
+    }
+    sk1.merge(sk2);
+    println(sk1.toString(true, true)); //L1 and above should be sorted in reverse. Ignore L0.
+    final int lvl1size = sk1.levelsArr[2] - sk1.levelsArr[1];
+    final QuantilesGenericSketchIterator<String> itr = sk1.iterator();
+    itr.next();
+    int prev = Integer.parseInt(itr.getQuantile().trim());
+    for (int i = 1; i < lvl1size; i++) {
+      if (itr.next()) {
+        int v = Integer.parseInt(itr.getQuantile().trim());
```
[the remark this answers is not in the document]
Comment (leerho) [reply src/test/java/org/apache/datasketches/kll/KllItemsSketchTest.java:755]: will fix

File: src/test/java/org/apache/datasketches/kll/KllItemsSketchTest.java [hunk 3]
```diff
@@ -740,6 +730,35 @@
...
+      final int j = rand.nextInt(n) + 1;
+      sk1.update(getString(j, 3));
+      sk2.update(getString(j +100, 3));
+    }
+    sk1.merge(sk2);
+    println(sk1.toString(true, true)); //L1 and above should be sorted in reverse. Ignore L0.
+    final int lvl1size = sk1.levelsArr[2] - sk1.levelsArr[1];
+    final QuantilesGenericSketchIterator<String> itr = sk1.iterator();
+    itr.next();
+    int prev = Integer.parseInt(itr.getQuantile().trim());
```
[the remark this answers is not in the document]
Comment (leerho) [reply src/test/java/org/apache/datasketches/kll/KllItemsSketchTest.java:752]: will fix

**Review by AlexanderSaydakov — APPROVED** [review]

**Pull request merged**